doc/port_auto_neg/port-auto-negotiation-design.md

```diff
@@ -360,29 +373,57 @@ SONiC already defined two fields related to port speed setting: **speed**, **aut
+
+ | value | mode |     Description                                           |
+ |:-----:|:----:|:---------------------------------------------------------:|
+ | 2     | auto | Enable autoneg if applicable to the transceiver attached  |
+ | 1     | on   | Enable autoneg regardless of the transceiver type         |
+ | 0     | off  | Disable autoneg                                           |
+
+For deployment considerations, users are encouraged to use only **auto** and **off** for the **autoneg**, while the **on** mode is for test purpose, it should only be activated when the transceiver is not correctly identified by the pmon#xcvrd, and forcing autoneg enabled is intended.
 
 #### Application DB Enhancements
 
-The change in APP_DB is similar to CONFIG_DB. 3 new fields **adv_speeds**, **interface_type**, **adv_interface_types** will be added to **PORT_TABLE** table:
+The change in APP_DB is similar to CONFIG_DB. 4 new fields **adv_speeds**, **remote_adv_speeds**, **interface_type**, **adv_interface_types** will be added to **PORT_TABLE** table:
 
 	; Defines information for port configuration
-	key                     = PORT_TABLE:port_name           ; configuration of the port
+	key                     = PORT_TABLE:port_name  ; configuration of the port
 	; field                 = value
     ...
-	adv_speeds              = STRING                         ; advertised speed list
-	interface_type          = STRING                         ; interface type
-	adv_interface_types     = STRING                         ; advertised interface types
-
-Valid value of the new fields are the same as **PORT** table in CONFIG_DB.
+	adv_speeds              = STRING                ; advertised speed list
+	interface_type          = STRING                ; interface type
+	adv_interface_types     = STRING                ; advertised interface types
+	rmt_adv_speeds          = STRING                ; advertised speed list of the remote peer
+	xcvr_capabilities       = STRING                ; transceiver capabilities
+	xcvr_speeds             = STRING                ; transceiver valid speed list
+
+- adv_speeds:  
+Valid value is the same as **PORT** table in CONFIG_DB.
+- interface_type:  
+Valid value is the same as **PORT** table in CONFIG_DB.
+- adv_interface_types:  
+Valid value is the same as **PORT** table in CONFIG_DB.
+- rmt_adv_speeds  
+Valid value is the same as **adv_speeds**, except that it's reported by the remote peer.
+- xcvr_capabilities:  
+Transceiver capabilities provided by pmon#xcvrd, it's a set of capabilities separated by comma, where `LT` stands for "link training" and `AN` for "auto negotiation". For example: "AN,LT". See detail description in section [PMON xcvrd Consideration](#pmon-xcvrd-consideration)
```

Comment:
As discussed during HLD review when enabling AN for a transceiver that doesn't support capability we should throw error message during configuration and pass it only if force option is used. Please update the HLD to reflect the same.

Reply:
We may also need to cover the scenario where pmon is not up or pmon comes up later.

Reply:
Thanks, and it's done